openedx/core/djangoapps/profile_images/tests/test_views.py

```diff
@@ -0,0 +1,232 @@
+"""
+Test cases for the HTTP endpoints of the profile image api.
+"""
+from contextlib import closing
+import unittest
+
+import ddt
+from django.conf import settings
+from django.core.urlresolvers import reverse
+import mock
+from PIL import Image
+from rest_framework.test import APITestCase, APIClient
+
+from student.tests.factories import UserFactory
+
+from ...user_api.accounts.api import set_has_profile_image, get_profile_image_names
+from ...user_api.accounts.helpers import get_profile_image_storage
+from ..images import create_profile_images, ImageValidationError
+from .helpers import make_image_file
+
+TEST_PASSWORD = "test"
+
+
+class ProfileImageEndpointTestCase(APITestCase):
+    """
+    Base class / shared infrastructure for tests of profile_image "upload" and
+    "remove" endpoints.
+    """
+    # subclasses should override this with the name of the view under test, as
+    # per the urls.py configuration.
+    _view_name = None
+
+    def setUp(self):
+        super(ProfileImageEndpointTestCase, self).setUp()
+        self.user = UserFactory.create(password=TEST_PASSWORD)
+        # Ensure that parental controls don't apply to this user
+        self.user.profile.year_of_birth = 1980
+        self.user.profile.save()
+        self.url = reverse(self._view_name, kwargs={'username': self.user.username})
+        self.client.login(username=self.user.username, password=TEST_PASSWORD)
+        self.storage = get_profile_image_storage()
+        # this assertion is made here as a sanity check because all tests
+        # assume user.profile.has_profile_image is False by default
+        self.assertFalse(self.user.profile.has_profile_image)
+
+    def tearDown(self):
+        for name in get_profile_image_names(self.user.username).values():
```

Comment:
Call super.tearDown (or use the way of registering a cleanup function in setUp).